code/modules/mining/mine_items.dm

```diff
@@ -428,3 +431,21 @@
 	anchored = 1
 	layer = MOB_LAYER - 0.2
 	density = 0
+
+/obj/item/weapon/emptysandbag
+	name = "empty sandbag"
+	desc = "A bag to be filled with sand."
+	icon = 'icons/obj/items.dmi'
+	icon_state = "sandbag"
+	w_class = 1
+
+/obj/item/weapon/emptysandbag/attackby(obj/item/W, mob/user, params)
+	if(istype(W,/obj/item/weapon/ore/glass || /obj/item/weapon/ore/glass/basalt))
```

Comment:
Because your logic is incorrect, but by SHEER CHANCE falls into a different section of DM that /just by chance/ evaluates everything as you intended, it's literally only because path2 is a sub of path1 (/basalt of /glass), which is why I still called it structurally (and morally) wrong.